### PR TITLE
Added ability to get all jobs.

### PR DESCRIPTION
**Pull request by gdey**
Added URL end point to get all jobs, and their status in the sytem. The status of jobs returns the position and total number of possible states for the job.

### CODE REVIEW

File: atlante/atlante.go [hunk 1]
```diff
@@ -33,9 +33,12 @@ type ImgStruct struct {
 	intermediate bool
 	// Have we generated the file?
 	// This will allow us to generate the file only when requested
-	generated bool
-	lck       sync.Mutex
-	image     *image.Image
+	generated                 bool
+	lck                       sync.Mutex
+	image                     *image.Image
```
Comment (ARolek) [inline atlante/atlante.go]: image.Image is an interface. What do we benefit by making it a pointer here?

File: atlante/atlante.go [hunk 2]
```diff
@@ -33,9 +33,12 @@ type ImgStruct struct {
...
+	startGenerationCallBack   func()
```
Comment (ARolek) [inline atlante/atlante.go]: these are empty funcs so they can be any function. Maybe make the funcs a type and then set this property to the type?

File: atlante/server/coordinator/postgresql/postgresql.go [hunk 3]
```diff
@@ -501,6 +505,119 @@ ORDER BY jobstatus.id desc limit 1;
 	}, true
 }
 
+func genAllSQL(primarySQL, defaultSQL string, limit uint) (string, error) {
```
Comment (ARolek) [inline atlante/server/coordinator/postgresql/postgresql.go]: can we get a comment on this func. The name and associated code are not clear to me.

File: atlante/server/server.go [hunk 4]
```diff
@@ -550,6 +550,18 @@ func (s *Server) JobInfoHandler(w http.ResponseWriter, request *http.Request, ur
 	}
 
 }
+func (s *Server) JobsHandler(w http.ResponseWriter, request *http.Request, urlParams map[string]string) {
+	// Hardcode 100 limit for now.
+	jobs, err := s.Coordinator.Jobs(100)
```
Comment (ARolek) [inline atlante/server/server.go]: move hard code value to a const?

File: atlante/atlante.go (hunk 1 above)
Comment (gdey) [reply atlante/atlante.go]: image is not the standard go library image here. It's the mbgl/image library. Image is a concrete type.  We talked about this before. :P

File: atlante/atlante.go (hunk 2 above)
Comment (gdey) [reply atlante/atlante.go]: I debated about this. Do we really need a type for such a simple call back function?

Comment (ARolek) [reply atlante/atlante.go]: maybe not, and I guess if it's not accepting anything it's unnecessary.

**Pull request merged**